library/ssl_msg.c

```diff
@@ -5865,6 +5865,55 @@ int mbedtls_ssl_read(mbedtls_ssl_context *ssl, unsigned char *buf, size_t len)
     return ret;
 }
 
+
+#if defined(MBEDTLS_SSL_SRV_C) && defined(MBEDTLS_SSL_EARLY_DATA)
+int mbedtls_ssl_read_early_data(mbedtls_ssl_context *ssl,
+                                unsigned char *buf, size_t len)
+{
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    const struct mbedtls_ssl_config *conf;
+    unsigned char *p = buf;
+
+    if (ssl == NULL || ((conf = ssl->conf) == NULL)) {
+        return MBEDTLS_ERR_SSL_BAD_INPUT_DATA;
+    }
+
+    if ((!mbedtls_ssl_conf_is_tls13_enabled(conf)) ||
```

Comment:
This seems redundant with the check below: if TLS 1.3 is not enabled, then surely it will not have been negotiated? Or am I missing something?

Reply:
Yes, belt and braces: first check the configuration then the maximum version to be negotiated or he negotiated version. I'd prefer to keep it but it you prefer I can remove it as well.

Reply:
Function re-written, not relevant anymore.